### PR TITLE
Hott 4481 update rails to 7.1

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link

HOTT-4481

### What?

Update Rails from 7.0 to 7.1.2

### Why?
To keep the framework updated with new features and security patches.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,9 +4,9 @@ ruby File.read('.ruby-version').chomp
 
 # Server
 gem 'puma'
-gem 'rails', '~> 7.0'
+gem 'rails', '~> 7.1.2'
 
-gem 'routing-filter', github: 'svenfuchs/routing-filter'
+gem 'routing-filter', github: 'trade-tariff/routing-filter'
```
Comment (alexdesi) [inline Gemfile:9]: updating `routing-filter` with the one that fixes the 7.1  CSRF compatibility issue.

File: config/environments/production.rb [hunk 2]
```diff
@@ -39,16 +38,22 @@
   # config.action_cable.url = "wss://example.com/cable"
   # config.action_cable.allowed_request_origins = [ "http://example.com", /http:\/\/example.*/ ]
 
+  # Assume all access to the app is happening through a SSL-terminating reverse proxy.
+  # Can be used together with config.force_ssl for Strict-Transport-Security and secure cookies.
+  config.assume_ssl = true
```
Comment (willfish) [inline config/environments/production.rb:43]:
```suggestion
  config.assume_ssl = ENV.fetch('RAILS_ASSUME_SSL', 'true') == 'true'
```

Comment (willfish) [reply config/environments/production.rb:43]: You okay to update Makefile to include https://github.com/trade-tariff/trade-tariff-frontend/blob/main/Makefile#L8

File: config/initializers/content_security_policy.rb [hunk 3]
```diff
@@ -1,26 +1,25 @@
 # Be sure to restart your server when you modify this file.
 
-# Define an application-wide content security policy
-# For further information see the following documentation
-# https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
+# Define an application-wide content security policy.
+# See the Securing Rails Applications Guide for more information:
+# https://guides.rubyonrails.org/security.html#content-security-policy-header
 
-Rails.application.configure do
```
Comment (willfish) [inline config/initializers/content_security_policy.rb:7]: We'll want to reinstate this configuration

Comment (alexdesi) [reply config/initializers/content_security_policy.rb:7]: :+1:  Done. The only changes are the comments.

**Review by willfish — APPROVED** [review]

**Review by willfish — APPROVED** [review]

**Pull request merged**